packages/atoms/connect/ConnectButton.tsx

```diff
@@ -0,0 +1,68 @@
+import { cn } from "@/lib/utils";
+import { useState } from "react";
+import * as React from "react";
+
+import { Button } from "../src/components/ui/button";
+import type { AtomsGlobalConfigProps } from "../types";
+
+// we accept two props: buttonText & onButtonClick
+// buttonText is for displaying buttonText
+// onButtonClick is a a method to handle onClick for button
+// it accepts another object which contains a callback in case of any error
+
+interface ConnectButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  buttonText?: string;
+  icon?: JSX.Element;
+  onSuccess?: () => void;
+  onError: () => void;
+}
+
+export function ConnectButton({
+  buttonText,
+  onClick,
+  onSuccess,
+  onError,
+  className,
+  icon,
+}: ConnectButtonProps & AtomsGlobalConfigProps) {
+  const [isProcessing, setIsProcessing] = useState<boolean>(false);
+  const [errMsg, setErrMsg] = useState<string>("");
+
+  function handleSubmit(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
+    e.preventDefault();
+
+    try {
+      setIsProcessing(true);
+      onClick && onClick(e);
+
+      // if user wants to handle onSuccess inside handleClick then it makes no sense to have a separate handler
+      // otherwise only if the user explicitly passes an onSuccess handler this gets triggered
+      if (onSuccess) {
+        onSuccess();
+      }
+    } catch (error) {
+      setIsProcessing(false);
+      onError();
+      setErrMsg(error?.message);
+    }
+
+    setIsProcessing(false);
+  }
+
+  return (
+    <div>
+      <Button
+        className={cn(
+          "bg-default text-default dark:text-muted dark:bg-muted relative inline-flex h-9 items-center whitespace-nowrap rounded-md px-4 py-2.5 text-sm font-medium !shadow-none transition-colors disabled:cursor-not-allowed",
+          className
+        )}
+        type="button"
+        disabled={!isProcessing}
+        onClick={(event) => handleSubmit(event)}>
+        {icon && icon}
+        {buttonText || "Install App"}
+      </Button>
+      {!!errMsg && <span>{errMsg}</span>}
```

Comment:
Cal handbook mentions to use ternaries over &&: https://handbook.cal.com/engineering/best-practices/prefer-ternaries-over-short-circuiting-and-and-for-react-components

Here is article: https://www.crocoder.dev/blog/react-conditional-rendering/

Reply:
@supalarry  I was trying to convert them into a boolean value at first using the `!!` operator just like on line 65, should I go back to that? It's also a possible way according to this article https://www.crocoder.dev/blog/react-conditional-rendering/

Reply:
You are right, `!!` is fine